File: LeetCode/71.Simplify_Path/Mario/71.Simplify_Path.swift
class Solution {
    func simplifyPath(_ path: String) -> String {
        let arr = path.split(separator: "/").filter { $0 != "" }

        var stack: [String] = []

        for e in arr {
            if e == ".." {
                stack.popLast()
            } else if e != "." {
                stack.append(String(e))
            }
        }

        // print(stack)

        return "/" + stack.joined(separator: "/")
    }
}

File: LeetCode/71.Simplify_Path/71.Simplify_Path.py
#!/usr/bin/env python3


class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for e in path.split("/"):
            if e == "..":
                if stack:
                    stack.pop()
            elif e == "." or e == "":
                continue
            else:
                stack.append(e)

        return "/" + ("/").join(stack)
